AsyncDisplayKit/UIResponder+AsyncDisplayKit.m

```diff
@@ -0,0 +1,31 @@
+//
+//  UIResponder+AsyncDisplayKit.m
+//  AsyncDisplayKit
+//
+//  Created by Adlai Holler on 2/13/17.
+//  Copyright © 2017 Facebook. All rights reserved.
+//
+
+#import "UIResponder+AsyncDisplayKit.h"
+
+#import <AsyncDisplayKit/ASAssert.h>
+#import <AsyncDisplayKit/ASBaseDefines.h>
+#import <AsyncDisplayKit/ASResponderChainEnumerator.h>
+
+@implementation UIResponder (AsyncDisplayKit)
+
+- (__kindof UIViewController *)asdk_associatedViewController
+{
+  ASDisplayNodeAssertMainThread();
+  
+  for (UIResponder *responder in [self asdk_responderChainEnumerator]) {
+    UIViewController *vc = ASDynamicCast(responder, UIViewController);
```

Comment:
Hmm I'll leave the link out because the best I could do is a general link to an overview of the responder chain. But responder chain always goes `aView -> aView.superview -> aView.superview.viewController -> aView.superview.viewController.view.superview` etc, which is documented.